src/helpers/internal_viewport.ts

```diff
@@ -100,6 +100,10 @@ export class InternalViewport {
       height = Math.max(height, this.viewportHeight); // if the viewport grid size is smaller than its client height, return client height
     }
 
+    if (lastRowEnd + FOOTER_HEIGHT > height && !this.getters.isDashboard()) {
+      height += FOOTER_HEIGHT;
+    }
```

Comment:
Didn't find other ways to determine whether it is in dashboard mode, so I keep this getter.